### PR TITLE
BI-7487 Add insolvency company number comparator

### PR DESCRIPTION
**Pull request by dlgroves-ch**
* Compare company numbers with insolvency cases on Oracle and MongoDB.
* Aggregate insolvency case comparison results into a new comparison
group named Company insolvency.
* Generalise MongoDB distinct service wrapper route.

Resolves:
[BI-7487](https://companieshouse.atlassian.net/browse/BI-7487)

### CODE REVIEW

**Review by kinpang-CH — APPROVED** [review]

**Review by phudson-ch — APPROVED** [review]

**Review by cgregoryCH — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/reconciliation/insolvency/InsolvencyCompanyNumberCompareMongoDBOracle.java [hunk 1]
```diff
@@ -7,24 +7,24 @@
...
- * Trigger a comparison between companies with insolvency cases on Oracle and MongoDB.
+ * Trigger a comparison between companies with insolvency cases on MongoDB and Oracle.
  */
 @Component
-public class InsolvencyCompanyNumberCompare extends RouteBuilder {
+public class InsolvencyCompanyNumberCompareMongoDBOracle extends RouteBuilder {
     @Override
     public void configure() throws Exception {
-        from("{{endpoint.insolvency.company_number.timer}}")
+        from("{{endpoint.insolvency.company_number.mongo_oracle.timer}}")
```
Comment (phudson-ch) [inline src/main/java/uk/gov/companieshouse/reconciliation/insolvency/InsolvencyCompanyNumberCompareMongoDBOracle.java:16]: Incorrect period separator: `company_number.mongo_oracle` should read `company_number_mongo_oracle`

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -54,7 +54,7 @@ endpoint.company_name_mongo_alpha.timer=timer:company_name_mongo_alpha?repeatCou
 endpoint.company_status_mongo_primary.timer=timer:company_status_mongo_primary?repeatCount=1&delay=${COMPANY_STATUS_MONGO_PRIMARY_DELAY}
 endpoint.company_status_mongo_alpha.timer=timer:company_status_mongo_alpha?repeatCount=1&delay=${COMPANY_STATUS_MONGO_ALPHA_DELAY}
 endpoint.company_status_mongo_oracle.timer=timer:company_status_mongo_oracle?repeatCount=1&delay=${COMPANY_STATUS_MONGO_ORACLE_DELAY}
-endpoint.insolvency.company_number.timer=timer:insolvency_company_number?repeatCount=1&delay=${INSOLVENCY_COMPANY_NUMBER_MONGO_ORACLE_DELAY}
+endpoint.insolvency.company_number.mongo_oracle.timer=timer:insolvency_company_number?repeatCount=1&delay=${INSOLVENCY_COMPANY_NUMBER_MONGO_ORACLE_DELAY}
```
Comment (phudson-ch) [inline src/main/resources/application.properties:57]: Incorrect period separator: company_number.mongo_oracle should read company_number_mongo_oracle

File: src/test/resources/application-stubbed.properties [hunk 3]
```diff
@@ -48,7 +48,7 @@ endpoint.company_name_mongo_alpha.timer=direct:company_name_mongo_alpha_trigger
 endpoint.company_status_mongo_primary.timer=direct:company_status_mongo_primary_trigger
 endpoint.company_status_mongo_alpha.timer=direct:company_status_mongo_alpha_trigger
 endpoint.company_status_mongo_oracle.timer=direct:company_status_mongo_oracle_trigger
-endpoint.insolvency.company_number.timer=direct:insolvency_company_number_trigger
+endpoint.insolvency.company_number.mongo_oracle.timer=direct:insolvency_company_number_mongo_oracle_trigger
```
Comment (phudson-ch) [inline src/test/resources/application-stubbed.properties:51]: Incorrect period separator: company_number.mongo_oracle should read company_number_mongo_oracle

**Review by phudson-ch — CHANGES_REQUESTED** [review]
Period separator has been used instead of underscore.

**Review by phudson-ch — APPROVED** [review]

**Pull request merged**